### PR TITLE
added crypto feature

### PR DESCRIPTION
**Pull request by sabify**
Hello,

This feature implemented `CryptoRng` trait to be used mainly in `dev-dependencies`/`tests` which are in need of this trait.

### CODE REVIEW

File: rand_xorshift/src/lib.rs [hunk 1]
```diff
@@ -115,3 +117,6 @@ impl SeedableRng for XorShiftRng {
         })
     }
 }
+
+#[cfg(feature = "crypto")]
+impl CryptoRng for XorShiftRng {}
```
Comment (dhardy) [inline rand_xorshift/src/lib.rs:122]: This (like many of the others) is definitely not a cryptographic RNG...

**Review by dhardy — COMMENTED** [review]
Could explain your motivation? Because the only real point of the `CryptoRng` trait is documentation. We certainly never saw a need for this.

**Comment by sabify** [comment]
Sometimes we need to write predictable tests with these RNG libs that need to satisfy `CryptoRng` trait along with `RngCore`.
We believe that these libs are not cryptographic RNG and the use of `crypto` feature need to be documented.

**Comment by dhardy** [comment]
> We believe that these libs are not cryptographic RNG

If you take that perspective, then why care about the `CryptoRng` trait at all?

**Comment by sabify** [comment]
To write `predictable tests` for those functions/methods that have `CryptoRng` marker trait! 

```rust
// I want to write predictable test for this function and 
// set a fixed state for the PRNG and get the same output every time in my tests.
fn do_something<T: CryptoRng + RngCore>(rng: T) -> Result(...) {}
```

Would you mind to change `crypto` feature name to `crypto-dev`/`crypto-debug`?

**Comment by dhardy** [comment]
But you can still write predictable tests over functions requiring a `CryptoRng` if you use a `CryptoRng` (e.g. `ChaCha20`) for those tests. So I still don't see the need for this... or particularly want it (it's more complexity and could be confusing).

**Comment by sabify** [comment]
What would be the complexity and confusing part of this dev feature? (specially if we name it `crypto-dev` and document it.)

We can use other libs even if they are **not** basically rng, but it would be good to have this dev feature in these **rngs** too!

**Comment by dhardy** [comment]
@vks @newpavlov thoughts? I'm tempted just to reject this.

**Comment by newpavlov** [comment]
I also do not think we should use `CryptoRng` for non-cryptographic RNGs, otherwise it completely loses its meaning of a tool for preventing accidental mistakes in cryptographic code (e.g. the `crypto` feature may leak via incorrect configuration somewhere, thus allowing use of non-secure RNGs in a production code). As was correctly noted, if you want to deterministically test code which demands `CryptoRng`, then you simply should use CSPRNGs such as ChaCha.

**Comment by vks** [comment]
I also don't see the need for this. If for some reason you want to mock a crypto RNG without actually using one, you can just implement your own type, possibly wrapping a non-crypto RNG. Sure, this requires more boilerplate code, but I think this also makes it less likely to be misused.

**Pull request closed without merging**